extension/chrome/elements/compose-modules/compose-send-btn-module.ts

```diff
@@ -197,21 +199,26 @@ export class ComposeSendBtnModule extends ViewModule<ComposeView> {
   };
 
 
-  private doSendMsg = async (msg: SendableMsg) => {
+  private doSendMsgs = async (msgs: SendableMsg[]) => {
     // if this is a password-encrypted message, then we've already shown progress for uploading to backend
     // and this requests represents second half of uploadable effort. Else this represents all (no previous heavy requests)
     const progressRepresents = this.view.pwdOrPubkeyContainerModule.isVisible() ? 'SECOND-HALF' : 'EVERYTHING';
     let msgSentRes: GmailRes.GmailMsgSend;
-    try {
-      this.isSendMessageInProgress = true;
-      msgSentRes = await this.view.emailProvider.msgSend(msg, (p) => this.renderUploadProgress(p, progressRepresents));
-    } catch (e) {
-      if (msg.thread && ApiErr.isNotFound(e) && this.view.threadId) { // cannot send msg because threadId not found - eg user since deleted it
-        msg.thread = undefined;
+    for (const msg of msgs) {
+      try {
+        this.isSendMessageInProgress = true;
         msgSentRes = await this.view.emailProvider.msgSend(msg, (p) => this.renderUploadProgress(p, progressRepresents));
-      } else {
-        this.isSendMessageInProgress = false;
-        throw e;
+      } catch (e) {
+        if (msg.thread && ApiErr.isNotFound(e) && this.view.threadId) { // cannot send msg because threadId not found - eg user since deleted it
+          msg.thread = undefined;
+          msgSentRes = await this.view.emailProvider.msgSend(msg, (p) => this.renderUploadProgress(p, progressRepresents));
+        } else {
+          this.isSendMessageInProgress = false;
+          throw e;
+        }
+      }
+      if (msg.externalId) {
+        this.view.acctServer.messageGatewayUpdate(msg.externalId, msgSentRes.id).catch(Catch.reportErr);
       }
     }
     BrowserMsg.send.notificationShow(this.view.parentTabId, { notification: `Your ${this.view.isReplyBox ? 'reply' : 'message'} has been sent.` });
```

Comment:
This PR now sends a separate message per recipient.
However, the problem is now -- how to show this message in inbox-active-thread? We don't want to show multiple copies, do we? So we need to think of a mechanism to collapse multiple messages into one. Right, @tomholub ?

Reply:
> This PR now sends a separate message per recipient.

This was planned eventually - sending a separate email message per recipient, each with a separate download link. But that was for a different issue. I don't think we even have any issue filed for that yet on browser extension.

As for this issue as we are tackling now, I think we should still be sending single one message, and then make another API call that updates FES with regards to the message ID that it was sent with.

Reply:
I think you are doing two issues in one here.

Issue 1: (this issue) make another API call after a message was sent, binding the sent message email id to the FES uploaded message

Issue 2: (not this issue) send one message per recipient, each with a separate url

Let's stick only to issue one in this PR, which should be a lot easier to do and review. Then we'll discuss and do issue 2 separately.

Reply:
ok, created PR #4353